vcomponent/src/vcomponent_hdmi_cec.c

```diff
@@ -234,114 +121,20 @@ typedef struct
   unsigned short cp_port;
   char* cp_path;
   vCHdmiCec_hal_t * cec_hal;
+  ut_kvp_instance_t *profile_instance;
+  bool bOpened;
 } vCHdmiCec_t;
 
-/*Common Data types*/
-
-typedef struct
-{
-  char* str;
-  int val;
-} strVal_t;
-
 
 /*Global variables*/
 static vCHdmiCec_t* gVCHdmiCec = NULL;
```

Comment:
you've got an instance. You cannot have a global, that's the point of having an instance, you don't own it, the memory is controlled by the caller.

All your checks need to check if the input is yours or not...

Reply:
I agree, but the problem in this case is there are 2 handles and 2 interfaces. 1 for vcomponent and 1 for HdmiCec. 
1.  Right now, the HdmiCec hal functions are taking in a int handle which is a problem in 64bit platforms like in Linux PCs. When HdmiCecOpen provides a handle, the handle is lost in translation due to this explicit conversion of pointer to int. Hence the subsequent HdmiCec functions dont have a reliable way to associate the integer handle to a memory instance of the driver struct. 
2. Secondly, When HdmiCec functions are called, we need a way to check if the vComponent is initialised before allowing the driver methods. 
That is why i had to use a global instance of vcomponent and associate the HdmiCec handle inside that to ensure sanity.

Reply:
let's add a comment /* TODO: when the ` int handle` is upgrade to support a void * in the future. We should re-review if we want to use a Singleton or not at that time. */